src/main/java/pl/krakow/service/CustomerRepositoryImpl.java

```diff
@@ -0,0 +1,56 @@
+package pl.krakow.service;
+
+import org.springframework.stereotype.Service;
+import pl.krakow.domain.Customer;
+import pl.krakow.exception.CustomerExistException;
+import pl.krakow.exception.CustomerNotFoundException;
+
+import java.util.List;
+
+@Service
+public class CustomerRepositoryImpl implements CustomerRepository {
+
+    private List<Customer> data;
+
+    public CustomerRepositoryImpl(List<Customer> data) {
```

Comment:
Ok, what if I ask you to support several different DB (H2, Faketest, MySql/MariaDB, Oracle, Postgress)(to be specified upon application start)? How painful that would be given such _architecture_?

Reply:
done!